pkg/statefulset/statefulset_test.go

```diff
@@ -1 +1,165 @@
 package statefulset
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+//nolint:funlen
+func TestListInAllNamespaces(t *testing.T) {
+	int32Ptr := func(i int32) *int32 { return &i }
+
+	generateStatefulSet := func(name, namespace string) *appsv1.StatefulSet {
+		return &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+				Labels:    map[string]string{"demo": name},
+			},
+			Spec: appsv1.StatefulSetSpec{
+				Replicas: int32Ptr(1),
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": name,
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"app": name,
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "test-container",
+								Image: "test-image",
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	testCases := []struct {
+		statefulsetMap         map[string][]string
+		addToRuntimeObjects    bool
+		expectedCount          int
+		statefulsetListOptions []metav1.ListOptions
+		expectedError          bool
+		expectedErrorMsg       string
+		debug                  bool
+	}{
+		{
+			statefulsetMap: map[string][]string{
+				"one":   {"one-1", "one-2", "one-3"},
+				"two":   {"two-1", "two-2"},
+				"three": {"three-1"},
+			},
+			addToRuntimeObjects:    true,
+			expectedCount:          int(6),
+			statefulsetListOptions: []metav1.ListOptions{},
+		},
+		{
+			statefulsetMap: map[string][]string{
+				"one": {"one-1"},
+			},
+			addToRuntimeObjects:    true,
+			expectedCount:          int(1),
+			statefulsetListOptions: []metav1.ListOptions{},
+		},
+		{
+			statefulsetMap: map[string][]string{
+				"one":   {"one-1", "one-2", "one-3"},
+				"two":   {"one-1", "two-2"},
+				"three": {"one-1"},
+			},
+			addToRuntimeObjects: true,
```

Comment:
Do you want to consider adding a test where addToRuntimeObjects is set to false?

Reply:
That's what I was thinking about but haven't come up with any

Reply:
```
addToRuntimeObjects: false,
expectedCount: 0
```

Reply:
added, thanks